spec/forms/key_people_forms_spec.rb

```diff
@@ -21,36 +28,242 @@
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      context "when a dob_day is not in the correct range" do
+        before(:each) do
+          key_people_form.dob_day = "42"
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+    end
+
+    describe "#dob_month" do
+      context "when a dob_month meets the requirements" do
+        it "is valid" do
+          expect(key_people_form).to be_valid
+        end
+      end
+
+      context "when a dob_month is blank" do
+        before(:each) do
+          key_people_form.dob_month = ""
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      context "when a dob_month is not an integer" do
+        before(:each) do
+          key_people_form.dob_month = "9.75"
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      context "when a dob_month is not in the correct range" do
+        before(:each) do
+          key_people_form.dob_month = "13"
         end
 
         it "is not valid" do
           expect(key_people_form).to_not be_valid
         end
       end
     end
+
+    describe "#dob_year" do
+      context "when a dob_year meets the requirements" do
+        it "is valid" do
+          expect(key_people_form).to be_valid
+        end
+      end
+
+      context "when a dob_year is blank" do
+        before(:each) do
+          key_people_form.dob_year = ""
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      context "when a dob_year is not an integer" do
+        before(:each) do
+          key_people_form.dob_year = "3.14"
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      context "when a dob_year is not in the correct range" do
+        before(:each) do
+          key_people_form.dob_year = (Date.today + 1.year).year.to_i
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+    end
+
+    describe "#date_of_birth" do
+      context "when a date_of_birth meets the requirements" do
+        it "is valid" do
+          expect(key_people_form).to be_valid
+        end
+      end
+
+      context "when a date of birth is not a valid date" do
+        before(:each) do
+          key_people_form.date_of_birth = nil
+        end
+
+        it "is not valid" do
+          expect(key_people_form).to_not be_valid
+        end
+      end
+
+      shared_examples_for "age limits for key people" do |business_type, age_limit|
+        before(:each) do
+          key_people_form.business_type = business_type
+        end
+
+        it "should not be valid when at the age limit" do
```

Comment:
Am I not getting this? The description is *should not be valid when at the age limit* but the expectation is checking that it is valid.

Reply:
Nah, this is just me copying and pasting 😅